src/cascadia/TerminalControl/TSFInputControl.cpp

```diff
@@ -104,7 +104,7 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     {
         if (_editContext != nullptr)
         {
-            // _editContext.NotifyFocusLeave(); TODO GitHub #3645: Enabling causes IME to no longer show up, need to determine if required
+            _editContext.NotifyFocusLeave();

```

Comment:
Can you double check that the IME with multiple tabs still works as expected and after closing tabs?  I seem to recall the IME would get into a state where it would not longer appear.

Reply:
Do you mean this Window?
![image](https://user-images.githubusercontent.com/2256941/71928331-9d790c00-3197-11ea-8fa8-c16191db23fa.png)

If yes, then I believe it works correctly, albeit slightly different compared to `origin/master`:
The current Windows Store version closes the IME after inserting a character. The version I built myself from this branch behaves like other software (like Firefox) on the other hand and doesn't close the Window unless you defocus it.
If you do have the time I would of course be very happy if you (or maybe someone else) could verify my changes. 🙂

BTW This PR doesn't fix the above "Emoji-Window" for the Chinese language though: Neither the Store nor my Dev version inserts any characters if I click any buttons in the Window.

Reply:
P.S.: Oh and the Unicode support is incomplete of course as can be seen in the above screenshot, but that's another story I suppose.

Reply:
Definently sounds like a big improvement.  I'll pull down your branch and double check.

Reply:
I took a look, I built both debug and release versions of your change and am still seeing #3745 , I can't reproduce #3706 .  

#3745 doesn't reproduce on the initial character, but consecutive characters (shift+/) get eaten and then eventually displayed.  I'm in Chinese (Simplfied, China) Microsoft Pinyin input method mode.

I also didn't run into any issues with my original concerns around IME not rendering.

Overall, I think we are good with this change as it definently fixes #3645 and #3706.  I'm just not sure about #3745.  @zadjii-msft any chance you can double check what I'm seeing with #3745?

Reply:
If I can't get to it tonight (3pm PST), then I'll give it a shot in the morning

Reply:
I did a clean build of this branch (`bcz dbg`) and tried to reproduce #3745 again.
Just to ensure I'm doing the same as you, @philnach, and others:

When the Terminal App is focused, your taskbar looks like this right?
![image](https://user-images.githubusercontent.com/2256941/71935390-f43a1200-31a6-11ea-9ad3-15ca5e47bf13.png)

And pressing the lone shift key (which is to the left of the right shift key), causes a `、` character to be printed?

In that case I'm not sure what I'm doing differently.
If I start a fresh instance of the App and press <kbd>Shift</kbd><kbd>/</kbd> 3 times, it prints ` ？？？` for me. 😕

Apart from that the following could make a difference:
* I'm using a German keyboard (instead of <kbd>/</kbd> it's a <kbd>-</kbd> for me), but if a Chinese input mode is used it should use the same virtual key code to a US one either way
* Fully updated Visual Studio 16.4.2
* I'm a Windows Insider on 19041.1

Reply:
Okay, I don't know if I'm doing it right, but I _maybe_ got #3745 to repro locally. I was changing input modes frequently and honestly I'm not sure I was doing it right at all. I'm still trying to make it happen again and I can't seem to. Maybe it has something to do with the shell? @philnach were you using `bash`, `cmd`, `pwsh`?

Reply:
Oh and btw @philnach: [This](https://drive.google.com/file/d/1xh-Y7j4l2Ns1oudhNoGu_3ESSFN4NhPj/view) is what I meant earlier when I said that the "Emoji-Window" doesn't insert anything when I use the Chinese input method. It seems to be slightly similar to the overarching "consecutive characters get eaten and then eventually displayed" issue.

Reply:
@zadjii-msft , I'm using cmd as the shell.

@lhecker , thanks for sharing the video and see the same behavior in the "Emoji-Window".  

Here's my setup:

- Fully updated VS 16.4.2

- Windows 10 19H1 (18363.535) 

Input method is the same as you sent in the picture:
![image](https://user-images.githubusercontent.com/19275540/71937524-7f90b480-3161-11ea-8024-20d4c9e7fa50.png)

I'm on a US keyboard.

**What I'm seeing:**
- Launch Release/x64 version of your branch.
- Set input method to Simplified Chinese.
- The first time I hit 'left shift + /' I get a ? 
- Second time I hit 'left shift + /' I get nothing new.  
- The third time I hit 'left shift + /', I finally get my second ?.

Here's what my prompt shows after each press of 'left shift + /'
1. ?
2. ?
3. ? ?
4. ? ?
5. ? ? ?

If I do the same thing with the same input method in Notpad, I have 5 questions marks after entering 'left shift + /' 5 times.

I double checked that the master branch behaves the same way as well 3 ? after 5 have been typed in.

Reply:
yikes, I'm on `10.0.19541.1001` and I can't get that to repro with the same steps. I'd say we leave #3745 open temporarily, just until we can make sure it goes away. @lhecker, do we have an issue tracking the Chinese IME + Emoji Picker issue？ If not, let's file one for a follow-up

Reply:
@zadjii-msft , thanks for double checking and odd that it doesn't repro.  I agree with leaving it temporarily open.

Reply:
@zadjii-msft --> https://github.com/microsoft/terminal/issues/4148
Please feel free to adjust the issue to your liking. 🙂